src/ILCompiler.Compiler/src/Compiler/SingleFileCompilationModuleGroup.cs

```diff
@@ -27,7 +27,11 @@ public override void AddCompilationRoots(IRootingServiceProvider rootProvider)
         {
             base.AddCompilationRoots(rootProvider);
 
-            AddCompilationRootsForExports((EcmaModule)_typeSystemContext.SystemModule, rootProvider);
+            foreach (var inputFile in _typeSystemContext.ReferenceFilePaths)
+            {
```

Comment:
I think it would be more appropriate to iterate over InputFilePaths instead.

The idea behind InputFilePaths and ReferenceFilePaths is that InputFilePaths is the source of roots - what is guaranteed to get compiled. ReferenceFilePaths can contain a lot of junk, but that it should not matter because of it will be touched and compiled in only if it gets actually references.